src/hostapi/wmme/pa_win_wmme.c

```diff
@@ -2329,6 +2326,8 @@ static PaError OpenStream( struct PaUtilHostApiRepresentation *hostApi,
     unsigned long outputDeviceCount = 0;                /* contains all devices and channel counts as local host api ids, even when PaWinMmeUseMultipleDevices is not used */
     char throttleProcessingThreadOnOverload = 1;
 
+    /* These are all the formats that can be represented in WAVEFORMATEX */
+    const PaSampleFormat nativeFormats = paUInt8 | paInt16 | paInt24 | paInt32 | paFloat32;
```

Comment:
Can this be gated on OS version? Before or after Vista+?

Reply:
Why? [Microsoft themselves say](https://learn.microsoft.com/en-us/windows-hardware/drivers/audio/background-of-non-pcm-support#waveout-api) that, all the way back on Windows 98 SE, MME already supported automatic conversions for all these formats through KMixer. I don't see any reason to believe this will break even on pre-Vista Windows.